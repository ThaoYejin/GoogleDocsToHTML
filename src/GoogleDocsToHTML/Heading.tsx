
import { docs_v1 } from "googleapis";
import React from "react";
import { TextRun } from "./TextRun";


export interface HTagProps {
    paraObj?: docs_v1.Schema$Paragraph;
}

export const Heading: React.FC<HTagProps> = ({
    paraObj
}) => {
    var hType = paraObj?.paragraphStyle?.namedStyleType;
    var content:React.ReactNode[] = [];
    paraObj?.elements?.forEach(function(element, index){
       content.push( <TextRun key={index} element={element}></TextRun>);
    })
     

    if (hType == "HEADING_1") {
        return (
            <h1>
                {content}
            </h1>
        )
    } else if (hType == "HEADING_2") {
        return (
            <h2>
                {content}
            </h2>
        )
    } else if (hType == "HEADING_3") {
        return (
            <h3>
                {content}
            </h3>
        )
    } else if (hType == "HEADING_4") {
        return (
            <h4>
                {content}
            </h4>
        )
    } else if (hType == "HEADING_5") {
        return (
            <h5>
                {content}
            </h5>
        )
    } else return (
        <h6>
            {content}
        </h6>
    );
}