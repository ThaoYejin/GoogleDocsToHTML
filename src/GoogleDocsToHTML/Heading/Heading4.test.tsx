import React from 'react';
import renderer from 'react-test-renderer';
import {Heading} from './Heading';

it('renders correctly', ()=>{
    var paraObj =  {
        "elements": [
          {
            "startIndex": 31,
            "endIndex": 41,
            "textRun": {
              "content": "Heading 4\n",
              "textStyle": {}
            }
          }
        ],
        "paragraphStyle": {
          "headingId": "h.3ojhkb4mt5h",
          "namedStyleType": "HEADING_4",
          "direction": "LEFT_TO_RIGHT"
        }
      };

    const tree = renderer
    .create(<Heading paraObj={paraObj}></Heading>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})