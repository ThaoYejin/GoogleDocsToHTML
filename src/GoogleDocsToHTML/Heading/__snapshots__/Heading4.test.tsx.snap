// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h4>
  <span
    style={Object {}}
  >
    Heading 4

  </span>
</h4>
`;
